### PR TITLE
docs: Fix untitled protos in API docs.

### PR DESCRIPTION
**Pull request by moderation**
Description: Fix untitled protos in API docs. Standardize on title casing.
Fixes: https://github.com/envoyproxy/envoy/issues/8525
Risk Level: Low
Testing: `bazel --nohome_rc test --config=libc++ //test/...`
Docs Changes: fix up some links to `deprecated.rst` to refer to internal links vs. Github
Release Notes: none required.

Signed-off-by: Michael P <michael@sooper.org>

### CODE REVIEW

File: tools/protodoc/protodoc.py [hunk 1]
```diff
@@ -202,6 +202,9 @@ def FormatHeaderFromFile(style, source_code_info, proto_name):
   if annotations.EXTENSION_ANNOTATION in source_code_info.file_level_annotations:
     extension = source_code_info.file_level_annotations[annotations.EXTENSION_ANNOTATION]
     formatted_extension = FormatExtension(extension)
+  if not annotations.DOC_TITLE_ANNOTATION in source_code_info.file_level_annotations and proto_name.startswith(
+      'envoy'):
+    raise ProtodocError('envoy API proto file missing title annotation: {}'.format(proto_name))
```
Comment (htuch) [inline tools/protodoc/protodoc.py]: Thanks for adding the check (and this awesome cleanup). I wonder if we should check the casing here as well? Could be a bit tricky for some cases, e.g. "gRPC status code".

Also, very tiny nit: "Envoy API proto" (the irony).

File: source/server/active_raw_udp_listener_config.cc [hunk 2]
```diff
@@ -3,7 +3,7 @@
 #include <memory>
 #include <string>
 
-#include "envoy/api/v2/listener/listener.pb.h"
+#include "envoy/api/v2/listener.pb.h"
```
Comment (htuch) [inline source/server/active_raw_udp_listener_config.cc]: Why all these changes? They seem unrelated to header casing.

File: api/envoy/extensions/filters/network/http_connection_manager/v3alpha/http_connection_manager.proto [hunk 3]
```diff
@@ -24,6 +24,7 @@ option java_outer_classname = "HttpConnectionManagerProto";
 option java_multiple_files = true;
 
 // [#protodoc-title: HTTP connection manager]
+
```
Comment (htuch) [inline api/envoy/extensions/filters/network/http_connection_manager/v3alpha/http_connection_manager.proto]: You probably want to verify if this has any measurable difference in docs. The first detached comment has special treatment I think.

Comment (mattklein123) [reply api/envoy/extensions/filters/network/http_connection_manager/v3alpha/http_connection_manager.proto]: +1, I think all of these need to be reverted. IIRC there can only be a single detached comment at the top of the file.

**Comment by mattklein123** [comment]
@moderation thank you so much for continuing my change! I can help you on Monday if there are formatting, etc. issues.
